x-pack/plugin/esql/compute/src/test/java/org/elasticsearch/compute/data/TestBlockBuilder.java

```diff
@@ -305,6 +319,9 @@ private static class TestBooleanBlockBuilder extends TestBlockBuilder {
 
         @Override
         public TestBlockBuilder appendObject(Object object) {
+            if (object instanceof Number number) {
+                object = number.intValue() % 2 == 0;
+            }
```

Comment:
This just allows us to use the same integer source values to create all block types, including boolean blocks ( BytesRet already does similar(ish) by just converting/wraping the values.